docs/section23/Rule23-19.md

```diff
@@ -0,0 +1,52 @@
+
+# Airside System - Rule 23-19  
+
+**Rule ID:** 23-19  
+**Rule Description:** Systems 5 - 8, the baseline system shall be modeled with preheat coils controlled to a fixed set point 20F less than the design room heating temperature setpoint.  
+**Rule Assertion:** B-RMR = expected value  
+**Appendix G Section:** Section 23 Air-side  
+**Appendix G Section Reference:** Section G3.1.3.19 Preheat Coils (Systems 5 through 8)  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-5, 6, 7, 8  
+
+**Manual Check:** None  
+**Evaluation Context:** Building  
+**Data Lookup:** None  
+**Function Call:**  
+
+1. baseline_hvac_type()  
+
+## Rule Logic:  
+
+- For each zone in B_RMR: `for zone_b in B_RMR...zones:`
+
+  - For each terminal in zone: `for terminal_b in zone_b.terminals:`
+
+    - Get HVAC system serving terminal: `hvac_b = terminal_b.served_by_heating_ventilation_air_conditioning_systems`
+  
+      - Check if HVAC system is type 5 through 8: `if baseline_hvac_type(hvac_b) in ["SYS-5", "SYS-6", "SYS-5", "SYS-8"]:`
+
+        - Set applicability flag: `rule_applicability_check = TRUE`
+
+        - Get zone design heating temperature setpoint: `zone_heating_setpoint_b = zone_b.design_thermostat_heating_setpoint`
+
+        - Save HVAC system and zone design heating temperature setpoint if temperature is higher than current value saved: `if zone_heating_setpoint_b > hvac_max_zone_setpoint_dict[hvac_b]: hvac_max_zone_setpoint_dict[hvac_b] = zone_heating_setpoint_b`
+
+- For each HVAC system that is type 5 through 8: `for hvac_b in hvac_max_zone_setpoint_dict.keys():`
+
+  - Get preheat system of HVAC system: `preheat_b = hvac_b.preheat_system`
+
+    **Rule Assertion:**
+
+    - Case 1: For each Baseline system type 5, 6, 7, or 8, if preheat coils is controlled to a fixed set point 20F less than the design room heating temperature setpoint: `if preheat_b.heating_coil_setpoint == hvac_max_zone_setpoint_dict[hvac_b] - 20: PASS`
+
```

Comment:
This RDS should also verify that preheat system is a coil, so I suggest to add ".... AND preheat_b.heating_system_type="FLUID LOOP"". 
I think we also need to verify that it references the right fluid loop.

Reply:
"AND preheat_b.heating_system_type="FLUID LOOP" is added. Also added FluidLoop.type == "Heating". Rule 21-18 will verify there is only one heating fluid loop.

Reply:
I do not understand what 'AND ( preheat_b.hot_water_loop )' is doing

Reply:
here preheat_b.heating_coil_setpoint == hvac_max_zone_setpoint_dict[hvac_b] - 20 ): PASS we assume that heating coil setpoint is in F. We should probably provide a unit conversion tip here, maybe heating_coil_setpoint[C to F]  or whatever convention we have previously followed.

Reply:
As Supriya noted above, I do not exactly follow what is "AND ( preheat_b.hot_water_loop)". Perhaps this is meant to check that the referenced loop  has FluidLoop.type == "Heating"?

Reply:
Correct. This is modified to check that it is connected to a heating hot water loop.